src/shogun/mathematics/graph/test/GraphTest.h

```diff
@@ -12,6 +13,37 @@ class GraphTest : public ::testing::Test
 	{
 	}
 
+	void test_binary_op_results(const std::shared_ptr<shogun::graph::Graph>& graph,
+		const shogun::SGVector<T>& X1,
+		const shogun::SGVector<T>& X2,
+		const shogun::SGVector<T>& expected_result1,
+		const shogun::SGVector<T>& expected_result2)
+	{
+		for (auto&& backend : this->m_backends)
+		{
+			graph->build(backend);
+
+			std::vector<std::shared_ptr<shogun::graph::Tensor>> result = graph->evaluate(
+				std::vector{std::make_shared<shogun::graph::Tensor>(X1),
+					std::make_shared<shogun::graph::Tensor>(X2)});
```

Comment:
it would be good to check the result size, as it should be always be 2 in a binary op.

Reply:
ah no, not because it is a binary, but because it has two outputs